### ISSUE TITLE
Path item components are ignored in `paths`

### ISSUE THREAD

**Issue by diego-aquino** [openapi-ts/openapi-typescript#1725]
**Description**

Hey! First of all, this lib has been great so far! However, it appears that path item components are being ignored in the type `paths`. If only path item components are used, `paths` is generated empty:

```ts
export type paths = Record<string, never>;
```

If I add one or more non-component path items, `paths` is generated with those, but component path items are still not included.

_[No error is shown]_

| Name                 | Version                |
| :------------------- | :--------------------- |
| `openapi-typescript` | `7.0.0`                |
| Node.js              | `20.15.0`              |
| OS + version         | `Linux Ubuntu + 22.04` |

> [!NOTE]
>
> **A bit of context**
> 
> I'm the creator of [Zimic](https://www.npmjs.com/package/zimic), an HTTP mocking library with extensive support to TypeScript. As part of our effort to provide type safety, I am integrating openapi-typescript with Zimic as part of https://github.com/zimicjs/zimic/issues/66. The idea is to generate the raw types using openapi-typescript and post-process them to follow Zimic's structure.

**Reproduction**

1. Create a file and paste the following schema, which uses path item components. This spec passes the Redocly validator and is valid according to [editor.swagger.io](https://editor-next.swagger.io/).

   <details>
   <summary><code>schema.yaml</code></summary>

   ```yaml
   openapi: 3.1.0

   info:
     title: Path Items
     version: 1.0.0
     license:
       name: MIT
       url: https://opensource.org/licenses/MIT

   servers:
     - url: https://my-example-server

   paths:
     /users:
       $ref: '#/components/pathItems/users'

   security:
     - bearerAuth: []

   components:
     pathItems:
       users:
         post:
           summary: Create user
           operationId: createUser

           requestBody:
             description: The user to create
             content:
               application/json:
                 schema:
                   type: object
                   properties:
                     name:
                       type: string

           responses:
             200:
               description: The user was created successfully
               content:
                 application/json:
                   schema:
                     type: object
                     properties:
                       id:
                         type: string
                       name:
                         type: string
             400:
               description: Bad request

     securitySchemes:
       bearerAuth:
         type: http
         scheme: bearer
         bearerFormat: JWT
   ```

   </details>

2. Generate the types:

   ```ts
   npx openapi-typescript schema.yaml -o schema.ts
   ```

**Current result**

`paths` ignores the component:

<details>
<summary><code>schema.ts</code></summary>

```ts
export type paths = Record<string, never>;
export type webhooks = Record<string, never>;
export interface components {
  schemas: never;
  responses: never;
  parameters: never;
  requestBodies: never;
  headers: never;
  pathItems: {
    users: {
      parameters: {
        query?: never;
        header?: never;
        path?: never;
        cookie?: never;
      };
      get?: never;
      put?: never;
      /** Create user */
      post: operations['createUser'];
      delete?: never;
      options?: never;
      head?: never;
      patch?: never;
      trace?: never;
    };
  };
}
export type $defs = Record<string, never>;
export interface operations {
  createUser: {
    parameters: {
      query?: never;
      header?: never;
      path?: never;
      cookie?: never;
    };
    /** The user to create */
    requestBody?: {
      content: {
        'application/json': {
          name?: string;
        };
      };
    };
    responses: {
      /** The user was created successfully */
      200: {
        headers: {
          [name: string]: unknown;
        };
        content: {
          'application/json': {
            id?: string;
            name?: string;
          };
        };
      };
      /** Bad request */
      400: {
        headers: {
          [name: string]: unknown;
        };
        content?: never;
      };
    };
  };
}
```

</details>

**Expected result**

`paths` should contain a reference to the component:

<details>
<summary><code>schema.ts</code></summary>

```ts
export interface paths {
  '/users': components['pathItems']['users'];
}
export type webhooks = Record<string, never>;
export interface components {
  schemas: never;
  responses: never;
  parameters: never;
  requestBodies: never;
  headers: never;
  pathItems: {
    users: {
      parameters: {
        query?: never;
        header?: never;
        path?: never;
        cookie?: never;
      };
      get?: never;
      put?: never;
      /** Create user */
      post: operations['createUser'];
      delete?: never;
      options?: never;
      head?: never;
      patch?: never;
      trace?: never;
    };
  };
}
export type $defs = Record<string, never>;
export interface operations {
  createUser: {
    parameters: {
      query?: never;
      header?: never;
      path?: never;
      cookie?: never;
    };
    /** The user to create */
    requestBody?: {
      content: {
        'application/json': {
          name?: string;
        };
      };
    };
    responses: {
      /** The user was created successfully */
      200: {
        headers: {
          [name: string]: unknown;
        };
        content: {
          'application/json': {
            id?: string;
            name?: string;
          };
        };
      };
      /** Bad request */
      400: {
        headers: {
          [name: string]: unknown;
        };
        content?: never;
      };
    };
  };
}
```

</details>

Let me know if this is an expected behavior or there's any problem with the schema I'm using.

**Checklist**

- [x] My OpenAPI schema passes the [Redocly validator](https://redocly.com/docs/cli/commands/lint/) (`npx @redocly/cli@latest lint`)
- [ ] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/openapi-ts/openapi-typescript/blob/main/packages/openapi-typescript/CONTRIBUTING.md))

**Comment by drwpow** [comment]
I’m surprised it took this long to catch this issue; this may have been present for a long time! We’re not testing for this. Would welcome a PR fixing this from someone (strong recommendation to add a failing test first since this issue has such a good repeatable example).
